pkg/kubelet/container/runtime.go

```diff
@@ -108,6 +108,11 @@ type Runtime interface {
 	// TODO: Change ContainerID to a Pod ID since the namespace is shared
 	// by all containers in the pod.
 	GetNetNS(containerID ContainerID) (string, error)
+	// Returns the container ID that represents the Pod, as passed to network
+	// plugins. For example if the runtime uses an infra container, returns
+	// the infra container's ContainerID.
+	// TODO: Change ContainerID to a Pod ID, see GetNetNS()
+	GetPodContainerID(*Pod) (*ContainerID, error)
```

Comment:
> WRT the larger question about returning network namespaces from setup, that's been discussed a bit in the context of the rkt runtime. I don't personally think kubelet should have anything to do with network namespaces since all it really cares about is the IP address(es) of the pod. Namespaces should be the domain of the network plugin. But we have a few steps to get there...
 
The way network plugins are setup currently kubelet acts as an intermediary between the runtime and network plugins. So it has to pass some "network context" information between the plugins and the runtime. This context IIUC is the infra container now. I was suggesting changing that to be something that is more portable. 

In any case, my comment can be converted to an issue for now since it needs major surgery.